### DISCUSSION TITLE
vscode-json-language-server schemas/schemastore

### DISCUSSION THREAD

**Discussion by ghost** [joaotavora/eglot#1410] [Show and tell]
I noticed there are no threads about vscode-json-languag-server jSON's schemastore support.

As a first step, here's a snippet that enables completions for package.json:

```elisp
(setq-default eglot-workspace-configuration (list (list :json
					:schemas [
						   (:fileMatch ["package.json"]
							       :url "https://json.schemastore.org/package.json")])))
```

Docs of vscode server: https://github.com/microsoft/vscode/blob/main/extensions/json-language-features/server/README.md

There's also eglot-yaml which supports schemastore (only for YAML, not json) https://github.com/yveszoundi/eglot-yaml/blob/main/eglot-yaml.el#L107-L111

Would be nice to have https://github.com/b0o/SchemaStore.nvim for Emacs

I also found this snippet: https://github.com/xhcoding/emacs.d/blob/a28d1ea98f85d5d58ed762dc6d0c88e5d9f919dc/README.org#json-ts-mode

**Comment by beetleman** [comment]
I done it in simplest/dummy way possible https://github.com/beetleman/.emacs.d/blob/0ffb8939e48e127777d8be3a25e6dbab5b31135c/init.el#L1055

1. download schemas from https://raw.githubusercontent.com/SchemaStore/schemastore/master/src/api/json/catalog.json and save in `.emacs.d/data`
2. read it
3. pass in `eglot-workspace-configuratio`

and it work for both, json and yaml

Its not code which should be used in `eglot` for sure but  its another snippet in this discussion:D

**Comment by peterhoeg** [comment]
Thanks for this @beetleman. It also works with `toml` if you add that in exactly like `yaml`.
